### ISSUE TITLE
New Resources: CodeStar Notifications

### ISSUE THREAD

**Issue by mwarkentin** [terraform-providers/terraform-provider-aws#10792]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

AWS just launched a new notification service for CodeStar (aka CodeBuild, CodePipeline, CodeDeploy, and CodeCommit).

Resources:

* https://aws.amazon.com/about-aws/whats-new/2019/11/introducing-notifications-for-aws-codecommit-aws-codebuild-aws-codedeploy-and-asw-codepipeline/
* https://docs.aws.amazon.com/codestar-notifications/latest/APIReference/Welcome.html
* Cloudformation resource: https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-codestarnotifications-notificationrule.html

### New or Affected Resource(s)

Guessing at some resources based on API docs:

* `aws_codestar_notification_rule`
* `aws_codestar_notification_target`

### Potential Terraform Configuration

```hcl
resource "aws_sns_topic" "codepipeline" {
  name = "codepipeline-notifications"
}

resource "aws_codestar_notification_rule" "codepipeline" {
  detail_type = "FULL"
  status = "enabled"

  event_type_ids = [
    "codepipeline-pipeline-pipeline-execution-failed",
    "codepipeline-pipeline-pipeline-execution-canceled",
    "codepipeline-pipeline-pipeline-execution-started",
    "codepipeline-pipeline-pipeline-execution-resumed",
    "codepipeline-pipeline-pipeline-execution-succeeded",
    "codepipeline-pipeline-pipeline-execution-superseded",
  ]

  targets = [
    "${aws_sns_topic.codepipeline.arn}" 
  ]

  tags = ...
}
```

**Comment by ewbankkit** [comment]
Support for CodeStar notifications added in [AWS SDK v1.25.27](https://github.com/aws/aws-sdk-go/releases/tag/v1.25.27),
Requires:
* https://github.com/terraform-providers/terraform-provider-aws/pull/10717

**Comment by bflad** [comment]
Support for a new `aws_codestartnotifications_notification_rule` resource has been merged and will release with version 2.49.0 of the Terraform AWS Provider, Thursday this week. Thanks to @spirius for the implementation. 👍
